modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItImplicitCastsTest.java

```diff
@@ -46,7 +54,10 @@ public void dropTables() {
     public void testFilter(ColumnPair columnPair) {
         prepareTables(columnPair);
 
-        assertQuery("SELECT T11.c2 FROM T11 WHERE T11.c2 > 1.0").check();
+        String value = columnPair.lhsLiteral(0);
+        String query = format("SELECT T11.c2 FROM T11 WHERE T11.c2 > CAST({} AS {})", value, columnPair.rhs);
```

Comment:
`CAST({} AS {})` -- this doesn't look like _implicit_ cast anymore

Reply:
An implicit cast is added to the left side of an expression ( T11.c2.)